CameraConfig.background yaml doesn't successfully parse in python

### What happened?

The `CameraConfig` structure has a `geometry::Rgba`-valued member called `background`.

As outlined in [this stackoverflow](https://stackoverflow.com/questions/79398649/specify-background-color-in-scenario-camera-config-yaml/79400212?noredirect=1#comment140027250_79400212) it can't really be set. The `!Rgba` token fails and setting it directly with a list of 3-4 floats also doesn't work.  For example:

```py
        yaml = "background: [0.1, 0.2, 0.3, 0.4]"
        dut = yaml_load_typed(schema=mut.CameraConfig, data=yaml)
```

Produces the error message (same as reported in stackoverflow):
```
Traceback (most recent call last):
  File "/mnt/nobackup/cache/bazel/_bazel_seancurtis/c482a09839de82c7b8c8a508d26ab2c3/execroot/_main/bazel-out/k8-opt/bin/bindings/pydrake/systems/py/sensors_test.runfiles/_main/bindings/pydrake/systems/test/sensors_test.py", line 262, in test_camera_config_yaml
    dut = yaml_load_typed(schema=mut.CameraConfig, data=yaml)
  File "/mnt/nobackup/cache/bazel/_bazel_seancurtis/c482a09839de82c7b8c8a508d26ab2c3/execroot/_main/bazel-out/k8-opt/bin/bindings/pydrake/systems/py/sensors_test.runfiles/_main/bindings/pydrake/common/yaml.py", line 648, in yaml_load_typed
    _merge_yaml_dict_into_target(
  File "/mnt/nobackup/cache/bazel/_bazel_seancurtis/c482a09839de82c7b8c8a508d26ab2c3/execroot/_main/bazel-out/k8-opt/bin/bindings/pydrake/systems/py/sensors_test.runfiles/_main/bindings/pydrake/common/yaml.py", line 567, in _merge_yaml_dict_into_target
    _merge_yaml_dict_item_into_target(
  File "/mnt/nobackup/cache/bazel/_bazel_seancurtis/c482a09839de82c7b8c8a508d26ab2c3/execroot/_main/bazel-out/k8-opt/bin/bindings/pydrake/systems/py/sensors_test.runfiles/_main/bindings/pydrake/common/yaml.py", line 515, in _merge_yaml_dict_item_into_target
    _merge_yaml_dict_into_target(
  File "/mnt/nobackup/cache/bazel/_bazel_seancurtis/c482a09839de82c7b8c8a508d26ab2c3/execroot/_main/bazel-out/k8-opt/bin/bindings/pydrake/systems/py/sensors_test.runfiles/_main/bindings/pydrake/common/yaml.py", line 528, in _merge_yaml_dict_into_target
    assert isinstance(yaml_dict, collections.abc.Mapping), yaml_dict
AssertionError: [0.1, 0.2, 0.3, 0.4]
```


### Version

_No response_

### What operating system are you using?

_No response_

### What installation option are you using?

_No response_

### Relevant log output

```shell

```

--- a/common/memory_file.h
+++ b/common/memory_file.h
@@ -85,16 +85,18 @@ class MemoryFile final {
    Serializing the %MemoryFile will produce such a string. Writing a yaml file
    by hand will be more challenging.
 
-   The following yaml would produce a %MemoryFile with contents equal to:
+   For this yaml:
 
-       This is an example of memory file test contents.
-
-   ```yaml
+   @code{yaml}
    contents: !!binary VGhpcyBpcyBhbiBleGFtcGxlIG9mIG1
            lbW9yeSBmaWxlIHRlc3QgY29udGVudHMu
    extension: .txt
    filename_hint: payload.txt
-   ```
+   @endcode
+
+   we would produce a %MemoryFile with contents equal to:
+
+       This is an example of memory file test contents.
    */
   template <typename Archive>
   void Serialize(Archive* a) {

--- a/geometry/rgba.h
+++ b/geometry/rgba.h
@@ -93,8 +93,8 @@ class Rgba {
 
   /** Passes this object to an Archive.
 
-   In YAML, an %Rgba is represented by an array-like list of three or four
-   numbers. E.g.,
+   In YAML, an %Rgba is represented by a mapping of the symbol `rgba` to
+   an array-like list of three or four numbers. E.g.,
 
         rgba: [0.5, 0.5, 1.0]
 
@@ -104,6 +104,29 @@ class Rgba {
 
    such that the first three values are red, green, and blue, respectively. If
    no fourth value is provided, alpha is defined a 1.0.
+
+   When another struct has an %Rgba-valued member (e.g.,
+   systems::sensors::CameraConfig::background), remember to include the full
+   mapping. For example, imagine the struct:
+
+   ```{cpp}
+   struct Foo {
+     Rgba color1;
+     Rgba color2;
+   };
+   ```
+
+   The correct yaml representation of this would be:
+
+   ```yaml
+   color1:
+     rgba: [0.5, 0.5, 1.0]
+   color2:
+     rgba: [1.0, 0.0, 0.0, 0.5]
+   ```
+
+   The *values* of `color1` and `color2` are the mapping from `rgba` to the
+   desired color tuples.
 
    Refer to @ref yaml_serialization "YAML Serialization" for background. */
   template <typename Archive>

